src/types/binary/word.ts

```diff
@@ -208,4 +210,58 @@ export class Word {
     const hexString = this.value.toString(16)
     return hexString.padStart(8, '0')
   }
+
+  /**
+   * throws exception if offset is not smaller than 32
+   * @param bitOffset 0-indexed offset
+   */
+  private checkIfBitOffsetInWordRange(bitOffset: number): void {
+    if (bitOffset >= Word.NUMBER_OF_BITS || bitOffset < 0) {
+      throw new VirtualBoardError(
+        'Offset is not within Word range',
+        VirtualBoardErrorType.BitOutOfTypeRange
+      )
+    }
+  }
+
+  /**
+   *
+   * @param bitOffset
+   * @returns
+   */
+  public isBitSet(bitOffset: number): boolean {
+    this.checkIfBitOffsetInWordRange(bitOffset)
+    return (this.value & (1 << bitOffset)) !== 0
+  }
+
+  /**
+   * sets the bit with 0-indexed offset from right side to 1
+   * @param bitOffset
+   * @returns new Word instance with changed value
+   */
+  public setBit(bitOffset: number): Word {
+    this.checkIfBitOffsetInWordRange(bitOffset)
+    return Word.fromSignedInteger(this.value | (1 << bitOffset))
```

Comment:
Shouldn't we use `fromUnsignedInteger` here and in the other methods for bits? Because according to the comment for `value` the number represents an unsigned integer so I would expect if we shift a unsigned integer we should also recreate a new Word again from an unsigned integer?

Reply:
If I change to `fromUnsignedInteger` then the test fail. I think it is if you set the highest bit (31 for Word) then it is no longer in the defined range of unsignedInteger. For Unsigned Integer negative values are allowed and so positive range is smaller than for signed integer (where only positive values exist)